src/routes/actions.js

```diff
@@ -38,31 +38,67 @@ class Actions {
     };
   }
 
+  /**
+   * @param {Number} recordId
+   */
+  async getRecord(recordId) {
+    return new this.implementation
+      .ResourceGetter(this.model, { recordId }).perform();
+  }
+
+  /**
+   * Generic hook controller.
+   *
+   * @param {Object} request the express.js request object
+   * @param {Object} response the express.js response object
+   * @param {Function} hook returns the response thanks to smartActionHook service
+   * @returns {*} the express.js response object
+   * @see getHookLoadController and getHookChangeController
+   */
+  async getHook(request, response, hook) {
+    const record = await this.getRecord(request.body.recordsId[0]);
+
+    try {
+      const updatedFields = await hook(record);
+
+      return response.status(200).send({ fields: updatedFields });
+    } catch ({ message }) {
+      this.logger.error('Error in smart action load hook: ', message);
+      return response.status(500).send({ message });
+    }
+  }
+
   /**
    * Generate a callback for express that handles the `load` hook.
    *
-   * @param {*} action The smart action
-   * @param {*} model The model of the smart action
-   * @param {*} Implementation Gives access to current Implementation (mongoose or sequelize)
+   * @param {Object} action The smart action
+   * @returns {Function} A route callback for express
+   */
+  getHookLoadController(action) {
+    return async (request, response) => (
+      this.getHook(request, response, async (record) => this.smartActionHook.getResponse(
+        action.hooks.load,
+        action.fields,
+        record,
+      )));
```

Comment:
I think I'm mistaking something.
The `getResponse` method signature is `getResponse(hook, record, fields)`.
Are `fields` and `record` inverted?

Same goes for change hook.

Reply:
Wow! Nice catch, thanks!! The "fun" part is that all test passed since everything is mocked 😱 

Fixed!